### PR TITLE
Improve test handling

### PR DESCRIPTION
**Pull request by jduck**
Add some flags to the test handling code to allow looking into them a bit easier -- especially in the case of a Heisenbug.

### CODE REVIEW

File: tests.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ die() {
 # Check for diff in system
 diff --help 2>&1 | grep -q gnu
 if [ "$?" = 0 ]; then
-  DIFF="diff --strip-trailing-cr"
+  DIFF="diff -a --strip-trailing-cr"
```
Comment (radare) [inline tests.sh]: what if the difference is in the color scheme? it will fail in nongnu and pass in gnu

**Comment by radare** [comment]
Looks good, but need to clarify this -a thing

**Comment by jduck** [comment]
It's fixing a real bug. That rcons problem I mentioned to you on IRC leads to (sometimes) a null byte being inserted into the output. This made diff think the output was binary and then only printed "binary files differ..."

I just checked OSX and its diff also has -a with the same meaning. Not sure I understand your GNU vs NON-GNU comment.

**Comment by radare** [comment]
Bht this trailing 0 can result in broken r2pipe and so on. This test must be fixed instead of hidden, which test does that?

**Comment by jduck** [comment]
I'm getting trailing junk on the "dbg_oo" test. Specifically, on the "dbg.ood 10" test.

```
Diff: --- /tmp/r2-regressions//dbg_oo-4qEBWs/exp        2016-05-17 11:03:51.804597334 -0500
+++ /tmp/r2-regressions//dbg_oo-4qEBWs/out      2016-05-17 11:03:52.060597341 -0500
@@ -1 +1 @@
-dbg:///bin/ls
+dbg:///bin/ls 
```

In this case there is a trailing space but before I saw a trailing null byte. I guess it just depends what the first character in the I.buffer is in rcons... I suspect this is a problem with rcons_stack_push/pop

**Comment by jduck** [comment]
BTW, nothing here hides this issue. In fact, it's actually making the issue more readily apparent (actual diff output is shown instead of just "Binary files differ")

**Comment by radare** [comment]
what about | hexdump -C to know why's this test failing exactly?

if you revert this change i'll merge the other commits and we can keep discussing about this change

**Comment by jduck** [comment]
Okay, I've just force pushed an update without that commit. Let's go to another PR or something =)

**Comment by radare** [comment]
Its ok for force push in prs. Github handles it properly

**Comment by jduck** [comment]
I already pushed. I didn't revert that commit but just removed it in an interactive rebase.

**Comment by jduck** [comment]
I can't PR the other stuff until this one is merged :-/

**Comment by radare** [comment]
thanks!

**Pull request merged**